lib/active_fulfillment/services/shipwire.rb

```diff
@@ -191,79 +191,88 @@ def parse_response(action, data)
-      response[:tracking_companies] = {}
-      response[:tracking_urls] = {}
-
-      document = REXML::Document.new(xml)
-      document.root.elements.each do |node|
-        if node.name == 'Order'
-          if node.attributes["shipped"] == "YES" && node.elements['TrackingNumber']
-            tracking_number = node.elements['TrackingNumber'].text.strip
-            response[:tracking_numbers][node.attributes['id']] = [tracking_number]
-
-            tracking_company = node.elements['TrackingNumber'].attributes['carrier']
-            response[:tracking_companies][node.attributes['id']] = [tracking_company.strip] if tracking_company
-
-            tracking_url = node.elements['TrackingNumber'].attributes['href']
-            response[:tracking_urls][node.attributes['id']] = [tracking_url.strip] if tracking_url
+      response = {
+        tracking_numbers: {},
+        tracking_companies: {},
+        tracking_urls: {}
+      }
+
+      Parsing.with_xml_document(xml, response) do |document, response|
+        document.root.try do |root_document|
+          root_document.elements.each do |node|
+            response[node.name.underscore.to_sym] = node.text.strip and next unless node.name == 'Order'.freeze && node.attributes['shipped'.freeze].text == 'YES'.freeze
+            node_tracking = node.at_css('TrackingNumber'.freeze)
+            unless node_tracking.nil?
+              node_id = node.attributes['id'.freeze].text.strip
+              tracking_number = node_tracking.text.strip
+              response[:tracking_numbers][node_id] = [tracking_number]
+
+              tracking_company = node_tracking.attributes['carrier'.freeze].try { |item| item.text.strip }
+              response[:tracking_companies][node_id] = [tracking_company] if tracking_company
+
+              tracking_url = node_tracking.attributes['href'.freeze].try { |item| item.text.strip }
+              response[:tracking_urls][node_id] = [tracking_url] if tracking_url
+            end
           end
-        else
-          response[node.name.underscore.to_sym] = text_content(node)
         end
       end
 
-      response[:success] = test? ? (response[:status] == '0' || response[:status] == 'Test') : response[:status] == '0'
-      response[:message] = response[:success] ? "Successfully received the tracking numbers" : message_from(response[:error_message])
+      response[:success] = test? ? (response[:status] == '0'.freeze || response[:status] == 'Test'.freeze) : response[:status] == '0'.freeze
+      response[:message] = response[:success] ? 'Successfully received the tracking numbers'.freeze : message_from(response[:error_message])
       response
     end
 
     def message_from(string)
       return if string.blank?
-      string.gsub("\n", '').squeeze(" ")
-    end
-
-    def text_content(xml_node)
-      text = xml_node.text
-      text = xml_node.cdatas.join if text.blank?
-      text
+      string.gsub("\n", ''.freeze).squeeze(' '.freeze)
```

Comment:
Chomp will remove more than new line. I'm not sure if we want to trim \r or not.

Here's from ruby doc

``
If $/ has not been changed from the default Ruby record separator, then chomp also removes carriage return characters (that is it will remove \n, \r, and \r\n). If $/ is an empty string, it will remove all trailing newlines from the string.
```